src/app/components/UserAssets/ConversionDialogContent.tsx

```diff
@@ -0,0 +1,115 @@
+import { getTokenContract } from '../../../utils/blockchain/contract-helpers';
+import { useTranslation } from 'react-i18next';
+import { useCanInteract } from '../../hooks/useCanInteract';
+import { useAccount } from '../../hooks/useAccount';
+import { useWeiAmount } from '../../hooks/useWeiAmount';
+import { bignumber } from 'mathjs';
+import { gasLimit } from '../../../utils/classifiers';
+import { TxType } from '../../../store/global/transactions-store/types';
+import { translations } from '../../../locales/i18n';
+import { FormGroup } from '../Form/FormGroup';
+import { AmountInput } from '../Form/AmountInput';
+import { BuyButton, Img } from './styled';
+import image from '../../../assets/images/arrow-down.svg';
+import { weiToFixed } from '../../../utils/blockchain/math-helpers';
+import { noop } from '../../constants';
+import { TxFeeCalculator } from '../../pages/MarginTradePage/components/TxFeeCalculator';
+
+interface IConversionDialogContentProps {
+  asset: Asset;
+  onClose: () => void;
+  onConversionSubmit: (value: string) => void;
+  convertTx: SendTxResponse;
+}
+
+export const ConversionDialogContent: React.FC<IConversionDialogContentProps> = ({
+  asset,
+  onConversionSubmit,
+  convertTx,
+}) => {
+  const contract = getTokenContract(asset);
+
+  const { t } = useTranslation();
+  const [amount, setAmount] = useState<string>('');
+  const connected = useCanInteract(true);
+  const account = useAccount();
+
+  const weiAmount = useWeiAmount(amount);
+
+  const isSubmitEnabled = useMemo(
+    () =>
+      connected && bignumber(weiAmount).greaterThan(0) && !convertTx.loading,
+    [connected, convertTx.loading, weiAmount],
+  );
+
+  const handleSubmit = useCallback(() => {
+    onConversionSubmit(weiAmount);
+  }, [onConversionSubmit, weiAmount]);
+
+  const txArgs = useMemo(() => [contract.address, weiAmount, account], [
+    account,
+    weiAmount,
+    contract,
+  ]);
+
+  const txConfig = useMemo(
+    () => ({
+      from: account,
+      gas: gasLimit[TxType.CONVERT_RUSDT_TO_XUSD],
+      nonce: 1, // it doesn't matter for the calculation
+    }),
+    [account],
+  );
+
+  return (
+    <>
+      <div className="tw-max-w-xs tw-mx-auto">
+        <div className="tw-text-2xl tw-text-center tw-font-semibold tw-mb-8">
+          {t(translations.userAssets.convertDialog.title)}
+        </div>
+
+        <FormGroup
+          label={`${t(translations.userAssets.convertDialog.from)}:`}
+          labelClassName="tw-text-sm"
+        >
+          <AmountInput
+            value={amount}
+            onChange={value => setAmount(value)}
+            asset={asset}
+          />
+        </FormGroup>
+
+        <Img src={image} alt="Arrow" />
+
+        <FormGroup
+          label={`${t(translations.userAssets.convertDialog.to)}:`}
+          labelClassName="tw-text-sm"
+        >
+          <AmountInput
+            value={weiToFixed(weiAmount, 6)}
+            onChange={noop}
```

Comment:
This is a bit weird. I know it wasn't developed in this story but `onChange` should be probably optional because it doesn't make any sense to have `onChange` defined when the input is `readonly`.

If it was too much of a work leave it as it is but if not, I'd rather see it optional.

Reply:
Yep, however it's currently required prop on this component and one of child components also depends on it. Need to add new task (if there is no on redesign epic yet) to refactor form elements a little bit.